Library/Homebrew/test/migrator_spec.rb

```diff
@@ -0,0 +1,269 @@
+    end
+
+    specify "#update_tabs" do
+      (new_keg_record/"bin").mkpath
+      tab = Tab.empty
+      tab.tabfile = HOMEBREW_CELLAR/"newname/0.1/INSTALL_RECEIPT.json"
+      tab.source["path"] = "/path/that/must/be/changed/by/update_tabs"
+      tab.write
+      subject.update_tabs
+      expect(Tab.for_keg(new_keg_record).source["path"]).to eq(new_formula.path.to_s)
+    end
+
+    specify "#migrate" do
+      tab = Tab.empty
+      tab.tabfile = HOMEBREW_CELLAR/"oldname/0.1/INSTALL_RECEIPT.json"
+      tab.source["path"] = old_formula.path.to_s
+      tab.write
+
+      shutup do
+        subject.migrate
+      end
+
+      expect(new_keg_record).to exist
+      expect(old_keg_record.parent).to be_a_symlink
+      expect(HOMEBREW_LINKED_KEGS/"oldname").not_to exist
+      expect((HOMEBREW_LINKED_KEGS/"newname").realpath).to eq(new_keg_record.realpath)
+      expect(old_keg_record.realpath).to eq(new_keg_record.realpath)
+      expect((HOMEBREW_PREFIX/"opt/oldname").realpath).to eq(new_keg_record.realpath)
+      expect((HOMEBREW_CELLAR/"oldname").realpath).to eq(new_keg_record.parent.realpath)
+      expect((HOMEBREW_PINNED_KEGS/"newname").realpath).to eq(new_keg_record.realpath)
+      expect(Tab.for_keg(new_keg_record).source["path"]).to eq(new_formula.path.to_s)
+    end
+
+    specify "#unlinik_oldname_opt" do
+      new_keg_record.mkpath
+      old_opt_record = HOMEBREW_PREFIX/"opt/oldname"
+      old_opt_record.unlink if old_opt_record.symlink?
+      old_opt_record.make_relative_symlink(new_keg_record)
+      subject.unlink_oldname_opt
+      expect(old_opt_record).not_to be_a_symlink
+    end
+
+    specify "#unlink_oldname_cellar" do
+      new_keg_record.mkpath
+      keg.unlink
+      keg.uninstall
+      old_keg_record.parent.make_relative_symlink(new_keg_record.parent)
+      subject.unlink_oldname_cellar
+      expect(old_keg_record.parent).not_to be_a_symlink
+    end
+
+    specify "#backup_oldname_cellar" do
+      (new_keg_record/"bin").mkpath
+      keg.unlink
+      keg.uninstall
+      subject.backup_oldname_cellar
+      expect(old_keg_record.subdirs).not_to be_empty
+    end
+
+    specify "#backup_old_tabs" do
+      tab = Tab.empty
+      tab.tabfile = HOMEBREW_CELLAR/"oldname/0.1/INSTALL_RECEIPT.json"
+      tab.source["path"] = "/should/be/the/same"
+      tab.write
+      migrator = Migrator.new(new_formula)
+      tab.tabfile.delete
+      migrator.backup_old_tabs
+      expect(Tab.for_keg(old_keg_record).source["path"]).to eq("/should/be/the/same")
+    end
+
+    describe "#backup_oldname" do
+      after(:each) do
+        expect(old_keg_record.parent).to be_a_directory
+        expect(old_keg_record.parent.subdirs).not_to be_empty
+        expect(HOMEBREW_LINKED_KEGS/"oldname").to exist
+        expect(HOMEBREW_PREFIX/"opt/oldname").to exist
+        expect(HOMEBREW_PINNED_KEGS/"oldname").to be_a_symlink
+        expect(keg).to be_linked
+      end
+
+      context "when cellar exists" do
```

Comment:
I'm always skeptical of contexts that contain nothing but a single example. I think these could be refactored to just be, e.g. `it "backs up the old name when cellar exists"`